lib/view_component/compiler.rb

```diff
@@ -56,7 +56,7 @@ def compile(raise_errors: false)
 
         component_class.class_eval <<-RUBY, template[:path], -1
           def #{method_name}
-            @output_buffer = ActionView::OutputBuffer.new
+            @output_buffer = self.class.template_buffer_instance
```

Comment:
Thoughts on removing this line, and initializing `@output_buffer` in the `ViewComponent::Base#render_in?` I think that may help your use-case by de-coupling the compiled method from `ActionView::OutputBuffer.new`

Reply:
@BlakeWilliams I think that could work, but do we want the base class to be concerned with setting up template/rendering minutia, vs the template compiler?

Reply:
Sorry for the delay on this. My current thinking is that `render_in` is a Rails specific API. Given that, it could make sense that `render_in` should be responsible for setting up the correct `@output_buffer` instance, if it's supposed to be configurable.

Reply:
I was curious how HAML handles this, and it looks like it brings its own output buffer instead of using `@output_buffer`. 

Here's the generated template code expanded a bit:

```ruby
  begin
    extend Haml::Helpers
    _hamlout = @haml_buffer = Haml::Buffer.new(haml_buffer, {})
    _erbout = _hamlout.buffer
    @output_buffer = output_buffer ||= ActionView::OutputBuffer.new if defined?(ActionView::OutputBuffer)
    _hamlout.buffer << ("<div>\n".freeze)

    _hamlout.buffer << ((
      ::Haml::Helpers.html_escape((( content
                                   )).to_s)
    ).to_s)

    _hamlout.buffer << ("\n".freeze)

    _hamlout.buffer << ((::Haml::Helpers.html_escape((( @message
                                                      )).to_s)
                        ).to_s)

    _hamlout.buffer << ("\n</div>\n".freeze)
    ::Haml::Util.html_safe(_erbout)
  ensure
    @haml_buffer = @haml_buffer.upper if @haml_buffer
  end
```

I did some quick verifying, and this works because we rely on the return value of the template, and not `@output_buffer`, which is empty at render time for HAML templates. Maybe that eliminates the need for using `@output_buffer` at all, at least for view components?

Reply:
@BlakeWilliams Sorry, I missed seeing a notification of this thread. I'd be cool with relocating the buffer instantiation to `render_in`. Is that something you'd like me to address in this PR? (BTW I didn't quite follow the point about how HAML's working.)

Reply:
Yeah let me revisit this…it's possible I may have already refactored out the need to change it.￼